### PR TITLE
Coupons - Search functionality

### PR DESCRIPTION
**Pull request by 0nko**
This PR adds the search (by string) functionality.

**To test:**
1. Go to Woo -> Coupons
2. Select a site
3. Tap on the Search coupons button
4. Enter a query that is present in some of the coupons
5. Notice the query results in coupons being fetched & saved in the database
6. Enter a query that isn't present
7. Notice no data is returned and database isn't changed

### CODE REVIEW

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/store/CouponStore.kt [hunk 1]
```diff
@@ -74,10 +74,40 @@ class CouponStore @Inject constructor(
...
+        return coroutineEngine.withDefaultContext(API, this, "searchCoupons") {
+            val response = restClient.fetchCoupons(site, page, pageSize, searchString)
+            when {
+                response.isError -> WooResult(response.error)
+                response.result != null -> {
+                    database.executeInTransaction {
+                        response.result.forEach { addCouponToDatabase(it, site) }
+                    }
+
+                    database
```
Comment (hichamboushaba) [inline plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/store/CouponStore.kt]: I think this dangling mention of the argument "database" here is not needed/useful.

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/store/CouponStore.kt [hunk 2]
```diff
@@ -74,10 +74,40 @@ class CouponStore @Inject constructor(
         }
     }
 
-    suspend fun fetchCoupon(
+    // Returns a boolean indicating whether more coupons can be fetched
```
Comment (hichamboushaba) [inline plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/store/CouponStore.kt]: This comment is not accurate I'm afraid, I think it was just mistakenly copied from `fetchCoupons`.

**Review by hichamboushaba — APPROVED** [review]
Nice work @0nko, left two np comments, but nothing is blocking.

File: plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/store/CouponStore.kt (hunk 2 above)
Comment (0nko) [reply plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/store/CouponStore.kt]: It was true but then I decided to return the actual search results, so the `Boolean` was replaced by `CouponSearchResult`. Thanks for noticing!

**Pull request merged**